src/main/java/com/ft/methodecontentcollectionmapper/mapping/EomFileToContentCollectionMapper.java

```diff
@@ -1,133 +1,181 @@
+	private static final String XPATH_POST_ID = "ObjectMetadata/WiresIndexing/ref_field";
+
+	private final DocumentStoreApiClient client;
+	private BlogUuidResolver blogUuidResolver;
+
+	public EomFileToContentCollectionMapper(DocumentStoreApiClient client, BlogUuidResolver blogUuidResolver) {
+		this.client = client;
+		this.blogUuidResolver = blogUuidResolver;
+	}
+
+	public ContentCollection mapPackage(EomFile eomFile, String transactionId, Date lastModified) {
+		try {
+			XPath xpath = XPathFactory.newInstance().newXPath();
+			Document attributesDocument = getDocumentBuilder()
+					.parse(new InputSource(new StringReader(eomFile.getAttributes())));
+			String webType = xpath.evaluate(XPATH_CONTAINER_WEB_TYPE, attributesDocument);
+
+			ContentCollectionType contentCollectionType = ContentCollectionType.fromWebType(webType, eomFile.getUuid());
+
+			return new ContentCollection.Builder().withUuid(eomFile.getUuid())
+					.withItems(extractItems(eomFile.getLinkedObjects(), transactionId))
+					.withPublishReference(transactionId).withLastModified(lastModified).withType(contentCollectionType)
+					.build();
+		} catch (ParserConfigurationException | SAXException | XPathExpressionException | IOException e) {
+			throw new TransformationException(e);
+		}
+	}
+
+	private List<Item> extractItems(final List<EomLinkedObject> linkedObjects, final String tid)
+			throws XPathExpressionException, ParserConfigurationException, SAXException, IOException {
+		if (linkedObjects == null) {
+			return Collections.emptyList();
+		}
+		
+		List<Item> items = new LinkedList<>();
+		for (final EomLinkedObject linkedObject : linkedObjects) {
+			Document attributesDocument = getLinkedObjectAttributes(linkedObject);
+			if (attributesDocument != null && isContentPlaceholder(attributesDocument)) {
+				items.add(resolveContentPlaceholder(attributesDocument, linkedObject.getUuid(), tid));
+			} else {
+				items.add(new Item.Builder().withUuid(linkedObject.getUuid()).build());
+			}
+		}
+
+		return items;
+	}
+
+	private Item resolveContentPlaceholder(Document attributesDocument, String linkedObjectUUID, String tid)
+			throws XPathExpressionException, ParserConfigurationException, SAXException, IOException {
+		final XPath xpath = XPathFactory.newInstance().newXPath();
+		String originalUUID = extractOriginalUUID(xpath, attributesDocument);
+
+		// this is a CPH referencing a blog
+		if (Strings.isNullOrEmpty(originalUUID)) {
+			return resolveToBlogItem(attributesDocument, linkedObjectUUID, tid);
+		}
+
+		// this is a CPH referencing an internal content
+		try {
+			client.canResolveUUID(originalUUID, tid);
+			return new Item.Builder().withUuid(originalUUID).build();
+		} catch (IllegalArgumentException | UuidResolverException e) {
+			throw new ContentCollectionMapperException(String.format("Unable to map content package"), e);
+		}
+	}
+
+	private Item resolveToBlogItem(Document attributesDocument, String linkedObjectUUID, String tid)
```

Comment:
There are some exceptions throws clause of this method that are never thrown in the method.